homeassistant/components/rotarex/config_flow.py

```diff
@@ -0,0 +1,47 @@
+import logging
+
+from rotarex_api import InvalidAuth, RotarexApi
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+class RotarexConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Rotarex."""
+
+    VERSION = 1
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            session = async_get_clientsession(self.hass)
+            api = RotarexApi(session)
+            try:
+                await api.login(user_input[CONF_EMAIL], user_input[CONF_PASSWORD])
+                await self.async_set_unique_id(user_input[CONF_EMAIL])
+                self._abort_if_unique_id_configured()
+                return self.async_create_entry(
+                    title=user_input[CONF_EMAIL], data=user_input
+                )
```

Comment:
This should be moved outside of the `try`. You can add a `else` block after the last `except` to put this in. https://docs.python.org/3/tutorial/errors.html#handling-exceptions (scroll down a little and you can see an example).